server-x/README.md

```diff
@@ -0,0 +1,39 @@
+# Keycloak.X Docker Image
+
+The `Dockerfile` for Keycloak.X Docker Image.
+
+See the `Build` section for more details on how to build the image.
+
+Once built, you can run the server in the same manner as when using the Keycloak.X distribution by passing any command-line argument:
+
+    docker run --name keycloak -p 8080:8080 \
+           -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=change_me \
+           IMAGE[:TAG] \
+           start-dev
+
+## Auto-Configuration
+
+Given the immutability of containers, you can use the `--auto-config` option in order to apply any configuration when running a new container. Once the container is created, subsequent restarts will never go through the configuration phase again but just start the server with the configuration previously defined.
```

Comment:
You won't be able to run the same command twice because the container is already created and Docker will complain. So, you would need to delete the container first or create a new one with a different name and new options.

If you don't use the `--auto-config` option the container will just run the `config` command. Containers are immutable and the arguments you used to create it are persisted into the container metadata. That is why we have this `--auto-config` option.

The recommended approach should be to create your own image from this one, where you can do all the configuration when building the image. The `--auto-config` is intended for helping to get a running server with different options without having to create your own image, with the cost that every time you run a new container, the config step will run and your container will take longer to start.

Reply:
> You won't be able to run the same command twice because the container is already created and Docker will complain. So, you would need to delete the container first or create a new one with a different name and new options.

Good point. Maybe this paragraph worth to be added exactly how you mention it as a NOTE/WARN to that documentation?

> The recommended approach should be to create your own image from this one, where you can do all the configuration when building the image.

Yes. So maybe it worth to add an example of this? Basically show how to configure the Keycloak with PostgreSQL with and without `--auto-config` . That is what I proposed to my comment below.

Again, maybe it is just me. My point is just to make the stuff easier to consume for the average user, who may not know what exactly is the difference between `config` and `start` and how container works under the covers etc :-)

Reply:
Sure, I'll work on your suggestions.

Reply:
@mposolda See the new `Extending the Image` section.